repos/spack_repo/builtin/packages/ripgrep/package.py

```diff
@@ -27,22 +27,37 @@ class Ripgrep(CargoPackage):
     version("13.0.0", sha256="0fb17aaf285b3eee8ddab17b833af1e190d73de317ff9648751ab0660d763ed2")
     version("11.0.2", sha256="0983861279936ada8bc7a6d5d663d590ad34eb44a44c75c2d6ccd0ab33490055")
 
+    variant("pcre2", default=False, description="Add support for Perl-style regex via PCRE2")
+
     depends_on("rust@1.85:", type="build", when="@15:")
-    depends_on("rust@1.72:", type="build", when="@14:")
+    depends_on("rust@1.72:", type="build", when="@14")
+    depends_on("rust@1.31:", type="build", when="@:13")
     depends_on("c", type="build")
 
+    with when("+pcre2"):
+        depends_on("pcre2")
+        depends_on("pkgconfig", type="build")
```

Comment:
```suggestion
    depends_on("pkgconfig", type="build", when="+pcre2")
    
    depends_on("pcre2", when="+pcre2")
```

Personal preference, but since there's only two dependencies related to `pcre2` support I'd just manually specify the when on both of them. (I'd also break up the sections so that the build dependencies are visually separate from the build/link dependencies.)

If additional features are added in the future that depend on `pkgconfig` it'll likely make sense to just make that build dependency required. However we'll keep it gated for now since there's only a single variant.

Reply:
Addressed with 9645980.